src/clusterfuzz/_internal/build_management/build_manager.py

```diff
@@ -425,27 +425,38 @@ def _unpack_build(self,
       _handle_unrecoverable_error_on_windows()
       return False
 
-    # Download build archive locally.
-    build_local_archive = os.path.join(build_dir, os.path.basename(build_url))
-
-    # Make the disk space necessary for the archive available.
-    archive_size = storage.get_object_size(build_url)
-    if archive_size is not None and not _make_space(archive_size,
-                                                    base_build_dir):
-      shell.clear_data_directories()
-      logs.log_fatal_and_exit(
-          'Failed to make space for download. '
-          'Cleared all data directories to free up space, exiting.')
-
-    logs.info(f'Downloading build from {build_url}.')
-    try:
-      storage.copy_file_from(build_url, build_local_archive)
-    except Exception as e:
-      logs.error(f'Unable to download build from {build_url}: {e}')
-      return False
+    should_download = not utils.is_chromium(
+    ) or not build_archive.unzip_over_http_compatible(build_url)
+
+    if should_download:
+      # Download build archive locally.
+      build_local_archive = os.path.join(build_dir, os.path.basename(build_url))
+
+      # Make the disk space necessary for the archive available.
+      archive_size = storage.get_object_size(build_url)
+      if archive_size is not None and not _make_space(archive_size,
+                                                      base_build_dir):
+        shell.clear_data_directories()
+        logs.log_fatal_and_exit(
+            'Failed to make space for download. '
+            'Cleared all data directories to free up space, exiting.')
+
+      logs.info(f'Downloading build from {build_url}.')
+      try:
+        storage.copy_file_from(build_url, build_local_archive)
+      except Exception as e:
+        logs.error(f'Unable to download build from {build_url}: {e}')
+        return False
+
+      opener = build_archive.open
+      file_arg = build_local_archive
+    else:
+      logs.info("Using an archive over HTTP, skipping archive download.")
+      opener = build_archive.open_uri
+      file_arg = build_url
```

Comment:
Consider extracting this to a helper method with early returns, or two helper methods - one for each branch.